bofire/data_models/features/categorical.py

```diff
@@ -360,37 +359,43 @@ class CategoricalOutput(Output):
     order_id: ClassVar[int] = 8
 
     categories: TCategoryVals
-    objective: Annotated[
-        List[Annotated[float, Field(type=float, ge=0, le=1)]], Field(min_items=2)
-    ]
+    objective: CategoricalObjective = Field(
+        default_factory=lambda: CategoricalObjective(w=1.0)
+    )
 
     @validator("categories")
     def validate_categories_unique(cls, categories):
         """validates that categories have unique names
 
         Args:
-            categories (List[str]): List of category names
+            categories (Union[List[str], Tuple[str]]): List or tuple of category names
 
         Raises:
             ValueError: when categories have non-unique names
 
         Returns:
-            List[str]: List of the categories
+            Tuple[str]: Tuple of the categories
         """
-        categories = list(categories)
         if len(categories) != len(set(categories)):
             raise ValueError("categories must be unique")
-        return categories
+        return tuple(categories)
 
     @validator("objective")
     def validate_objective(cls, objective, values):
-        if len(objective) != len(values["categories"]):
+        weights = objective.weights
+        if len(weights) != len(values["categories"]):
             raise ValueError("Length of objectives and categories do not match.")
-        for o in objective:
-            if o > 1:
-                raise ValueError("Objective values has to be smaller equal than 1.")
-            if o < 0:
-                raise ValueError("Objective values has to be larger equal than zero")
+        for w in weights:
```

Comment:
If we go for this structure, this validation should go to the inside of the objective class as validator there, or we just do it via the annotated stuff. But then also there.